test/Test/Toml/Parsing/Unit.hs

```diff
@@ -20,28 +22,38 @@ spec_Parser = do
       failOn p given            = parse p "" `shouldFailOn` given
       parseXSatisfies p given f = parse p "" given `parseSatisfies` f
 
-      parseArray   = parseX arrayP
-      parseBool    = parseX boolP
-      parseDouble  = parseX doubleP
-      parseInteger = parseX integerP
-      parseKey     = parseX keyP
-      parseKeyVal  = parseX keyValP
-      parseText    = parseX textP
-      parseTable   = parseX tableHeaderP
-      parseToml    = parseX tomlP
+      parseArray    = parseX arrayP
+      parseBool     = parseX boolP
+      parseDateTime = parseX dateTimeP
+      parseDouble   = parseX doubleP
+      parseInteger  = parseX integerP
+      parseKey      = parseX keyP
+      parseKeyVal   = parseX keyValP
+      parseText     = parseX textP
+      parseTable    = parseX tableHeaderP
+      parseToml     = parseX tomlP
 
-      arrayFailOn  = failOn arrayP
-      boolFailOn   = failOn boolP
-      doubleFailOn = failOn doubleP
-      keyValFailOn = failOn keyValP
-      textFailOn   = failOn textP
+      arrayFailOn    = failOn arrayP
+      boolFailOn     = failOn boolP
+      dateTimeFailOn = failOn dateTimeP
+      doubleFailOn   = failOn doubleP
+      keyValFailOn   = failOn keyValP
+      textFailOn     = failOn textP
 
       doubleSatisfies = parseXSatisfies doubleP
 
       quoteWith q t = q <> t <> q
       squote        = quoteWith "'"
       dquote        = quoteWith "\""
 
+      makeDay year month day            = Day $ fromGregorian year month day
+      makeHours hour minute second      = Hours $ TimeOfDay hour minute second
+      makeLocal (Day day) (Hours hours) = Local $ LocalTime day hours
+      makeLocal _          _            = undefined
```

Comment:
I don't see why pattern-matching is not exhaustive here... Anyway, it's better to avoid `undefined`. Or at least put `error`

Reply:
Since `LocalTime` accepts two arguments of type `Day` and `TimeOfDay` (or `Local` accepts `Day` and `Hours`), there can't be a result if `makeLocal` accepts arguments of different types, `Zoned` and `Local` for example.

I can still switch to using `error` instead of `undefined`.